lpcompiler.pas

```diff
@@ -1422,6 +1422,7 @@ function TLapeCompiler.ParseMethodHeader(out Name: lpString; addToScope: Boolean
       addVar(lptOut, Result.Res, 'Result');
     end;
 
+    Result.Name := Name;
```

Comment:
You set the name of the *type* to be equal to the name of the *function* here. Doesn't that cause issues?